src/gridfs.rs

```diff
@@ -0,0 +1,168 @@
+use core::task::{Context, Poll};
+use std::{
+    io::{self, Result},
+    marker::PhantomPinned,
+    pin::Pin,
+};
+
+use crate::{
+    concern::{ReadConcern, WriteConcern},
+    cursor::Cursor,
+    selection_criteria::ReadPreference,
+    Database,
+};
+
+use bson::{oid::ObjectId, DateTime, Document};
+use serde::Deserialize;
+use typed_builder::TypedBuilder;
+
+use tokio::io::{AsyncRead, AsyncWrite, ReadBuf};
+
+#[derive(Clone, Debug, Default, Deserialize, TypedBuilder)]
+#[builder(field_defaults(setter(into)))]
+#[non_exhaustive]
+pub struct GridFSBucketOptions {
+    bucket_name: Option<String>,
+    chunk_size_bytes: Option<i32>,
```

Comment:
Can we make all of the options fields `pub`? That way users can reassign fields as needed.